### PR TITLE
RESTWS-558 | Create a rest resource/api for org.openmrs.OrderSet

### PR DESCRIPTION
**Pull request by ys-achinta**
Added two resources orderSet,
      orderSetMember for saving, updating, deleting, orderSets.
    (The orderSetMember save/update is taken care by the save/update of orderSet)

### CODE REVIEW

**Comment by djazayeri** [comment]
At a very quick glance on my phone: shouldn't ordersetmember be a
subresource of orderset?

-Darius (by phone)

**Comment by ys-achinta** [comment]
@djazayeri 
 I have made all the changes incorporating the advise that you suggested.

Please have a look and let me know if I missed something.
Also the openmrs-core related changes are here:
https://github.com/openmrs/openmrs-core/pull/1752/commits/fd36c0ccd47d0a84bd5aa1a43a350d58fdf65c72

File: omod-1.12/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_12/OrderSetMemberResource1_12.java [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+
+    @Override
+    public OrderSetMember getByUniqueId(String uniqueId) {
+        return Context.getOrderSetService().getOrderSetMemberByUuid(uniqueId);
+    }
+
+    @Override
+    protected void delete(OrderSetMember orderSetMember, String reason, RequestContext context) throws ResponseException {
+        OrderSet orderSet = orderSetMember.getOrderSet();
+        orderSet.retireOrderSetMember(orderSetMember);
```
Comment (dkayiwa) [inline omod-1.12/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_12/OrderSetMemberResource1_12.java]: Do we intentionally ignore the retire reason?

File: omod-1.12/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_12/OrderSetMemberResource1_12.java [hunk 2]
```diff
@@ -0,0 +1,154 @@
+/**
+ * The contents of this file are subject to the OpenMRS Public License
+ * Version 1.0 (the "License"); you may not use this file except in
```
Comment (dkayiwa) [inline omod-1.12/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_12/OrderSetMemberResource1_12.java]: Should the license be version 2.0? :)

File: omod-1.12/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_12/OrderSetResource1_12.java [hunk 3]
```diff
@@ -0,0 +1,106 @@
+/**
+ * The contents of this file are subject to the OpenMRS Public License
+ * Version 1.0 (the "License"); you may not use this file except in
```
Comment (dkayiwa) [inline omod-1.12/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_12/OrderSetResource1_12.java]: Same as above for license version.

File: omod-1.12/src/test/java/org/openmrs/module/webservices/rest/web/RestTestConstants1_12.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
+/**
+ * The contents of this file are subject to the OpenMRS Public License
+ * Version 1.0 (the "License"); you may not use this file except in
```
Comment (dkayiwa) [inline omod-1.12/src/test/java/org/openmrs/module/webservices/rest/web/RestTestConstants1_12.java]: Same as above for license version.

File: omod-1.12/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs1_12/OrderSetController1_12Test.java [hunk 5]
```diff
@@ -0,0 +1,177 @@
+/**
+ * The contents of this file are subject to the OpenMRS Public License
+ * Version 1.0 (the "License"); you may not use this file except in
```
Comment (dkayiwa) [inline omod-1.12/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs1_12/OrderSetController1_12Test.java]: Same as above for license version.

File: omod-1.12/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs1_12/OrderSetMemberController1_12Test.java [hunk 6]
```diff
@@ -0,0 +1,141 @@
+package org.openmrs.module.webservices.rest.web.v1_0.controller.openmrs1_12;
```
Comment (dkayiwa) [inline omod-1.12/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs1_12/OrderSetMemberController1_12Test.java]: Did you just forget to add the license header to this file?

File: omod-1.12/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_12/OrderSetMemberResource1_12Test.java [hunk 7]
```diff
@@ -0,0 +1,54 @@
+package org.openmrs.module.webservices.rest.web.v1_0.resource.openmrs1_12;
```
Comment (dkayiwa) [inline omod-1.12/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_12/OrderSetMemberResource1_12Test.java]: Same here about missing license header.

File: omod-1.12/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_12/OrderSetResource1_12Test.java [hunk 8]
```diff
@@ -0,0 +1,63 @@
+/**
+ * The contents of this file are subject to the OpenMRS Public License Version
+ * 1.0 (the "License"); you may not use this file except in compliance with the
```
Comment (dkayiwa) [inline omod-1.12/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_12/OrderSetResource1_12Test.java]: Should be version 2.0

**Pull request merged**